src/osd/ReplicatedBackend.cc

```diff
@@ -1847,10 +1914,21 @@ bool ReplicatedBackend::handle_pull_response(
 	   << ", new progress " << pi.recovery_progress
 	   << dendl;
 
+  interval_set<uint64_t> data_zeros;
+  uint64_t z_offset = pop.before_progress.data_recovered_to;
+  uint64_t z_length = pop.after_progress.data_recovered_to - pop.before_progress.data_recovered_to;
+  if(z_length)
+    data_zeros.insert(z_offset, z_length);
```

Comment:
I don't really understand what you mean by that.  If this region doesn't necessarily include zeroes, then why is it called data_zeroes?